In [10]:
# Name Dipika Sharma
# DSC 550 T302 Data Mining
# Week 1

# Exercise 1.2

In [11]:
## Importing the required libraries

import pandas as pd
import numpy as np

### 1. Load the dataset as a Pandas data frame.

In [40]:
## Load the CSV dataset to Pandas data frame Video_Games_df.

Video_Games_df = pd.read_csv("Video_Games_Sales_as_at_22_Dec_2016.csv")

### 2. Display the first ten rows of data.

In [43]:
## to display the first ten rows of the dataframe using head command.

Video_Games_df.head(10)

Name Platform  Year_of_Release         Genre  \
0                 Wii Sports      Wii           2006.0        Sports   
1          Super Mario Bros.      NES           1985.0      Platform   
2             Mario Kart Wii      Wii           2008.0        Racing   
3          Wii Sports Resort      Wii           2009.0        Sports   
4   Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing   
5                     Tetris       GB           1989.0        Puzzle   
6      New Super Mario Bros.       DS           2006.0      Platform   
7                   Wii Play      Wii           2006.0          Misc   
8  New Super Mario Bros. Wii      Wii           2009.0      Platform   
9                  Duck Hunt      NES           1984.0       Shooter   

  Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  \
0  Nintendo     41.36     28.96      3.77         8.45         82.53   
1  Nintendo     29.08      3.58      6.81         0.77         40.24   
2  Nintendo     15.68     12.76      3.79         3.29         35.52   
3  Nintendo     15.61     10.93      3.28         2.95         32.77   
4  Nintendo     11.27      8.89     10.22         1.00         31.37   
5  Nintendo     23.20      2.26      4.22         0.58         30.26   
6  Nintendo     11.28      9.14      6.50         2.88         29.80   
7  Nintendo     13.96      9.18      2.93         2.84         28.92   
8  Nintendo     14.44      6.94      4.70         2.24         28.32   
9  Nintendo     26.93      0.63      0.28         0.47         28.31   

   Critic_Score  Critic_Count User_Score  User_Count Developer Rating  
0          76.0          51.0          8       322.0  Nintendo      E  
1           NaN           NaN        NaN         NaN       NaN    NaN  
2          82.0          73.0        8.3       709.0  Nintendo      E  
3          80.0          73.0          8       192.0  Nintendo      E  
4           NaN           NaN        NaN         NaN       NaN    NaN  
5           NaN           NaN        NaN         NaN       NaN    NaN  
6          89.0          65.0        8.5       431.0  Nintendo      E  
7          58.0          41.0        6.6       129.0  Nintendo      E  
8          87.0          80.0        8.4       594.0  Nintendo      E  
9           NaN           NaN        NaN         NaN       NaN    NaN

### 3. Find the dimensions (number of rows and columns) in the data frame. What do these two numbers represent in the context of the data?

In [44]:
print("The number of rows and columns in the data frame is: {}".format(Video_Games_df.shape))

The number of rows and columns in the data frame is: (16719, 16)


##### The shape command is used to tell us the number of rows and columns in the data frame. Looking at the above result we can see that the dataframe we using has 16719 rows and 16 columns.

### 4. Find the top five games by critic score.

In [45]:
## To get the top five games sorting the data using sort_value by cristic_score.
## Creating new dataframe and Sort the data in descending order.
## Display the top 5 games using the head command in new data frame.

Games_Top5_df = Video_Games_df.sort_values(by=['Critic_Score'], ascending=False)
Games_Top5_df.head(5)

Name Platform  Year_of_Release     Genre  \
227   Tony Hawk's Pro Skater 2       PS           2000.0    Sports   
57         Grand Theft Auto IV      PS3           2008.0    Action   
51         Grand Theft Auto IV     X360           2008.0    Action   
5350               SoulCalibur       DC           1999.0  Fighting   
165         Grand Theft Auto V     XOne           2014.0    Action   

                 Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
227             Activision      3.05      1.41      0.02         0.20   
57    Take-Two Interactive      4.76      3.69      0.44         1.61   
51    Take-Two Interactive      6.76      3.07      0.14         1.03   
5350    Namco Bandai Games      0.00      0.00      0.34         0.00   
165   Take-Two Interactive      2.81      2.19      0.00         0.47   

      Global_Sales  Critic_Score  Critic_Count User_Score  User_Count  \
227           4.68          98.0          19.0        7.7       299.0   
57           10.50          98.0          64.0        7.5      2833.0   
51           11.01          98.0          86.0        7.9      2951.0   
5350          0.34          98.0          24.0        8.8       200.0   
165           5.48          97.0          14.0        7.9       764.0   

                    Developer Rating  
227   Neversoft Entertainment      T  
57             Rockstar North      M  
51             Rockstar North      M  
5350                    Namco      T  
165            Rockstar North      M

### 5. Find the number of video games in the data frame in each genre.

In [47]:
## Create new dataframe with only two columns 'Name' and 'Genre' of existing dataframe.

## Use the groupby function on new dataframe to display the number of video games in each genre.

VGames_df = Video_Games_df[['Name','Genre']]
VGames_df.groupby(['Genre']).agg(['count'])

Name
             count
Genre             
Action        3370
Adventure     1303
Fighting       849
Misc          1750
Platform       888
Puzzle         580
Racing        1249
Role-Playing  1500
Shooter       1323
Simulation     874
Sports        2348
Strategy       683

### 6. Find the first five games in the data frame on the SNES platform.

In [48]:
## Creating new dataframe with games only on the SNES platform.
## Use head command to display the top 5 games in the dataframe.

VGames_SNES_df = Video_Games_df.loc[Video_Games_df['Platform'] == "SNES"] 
VGames_SNES_df.head(5)

Name Platform  Year_of_Release     Genre  \
18                      Super Mario World     SNES           1990.0  Platform   
56                  Super Mario All-Stars     SNES           1993.0  Platform   
71                    Donkey Kong Country     SNES           1994.0  Platform   
76                       Super Mario Kart     SNES           1992.0    Racing   
137  Street Fighter II: The World Warrior     SNES           1992.0  Fighting   

    Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  \
18   Nintendo     12.78      3.75      3.54         0.55         20.61   
56   Nintendo      5.99      2.15      2.12         0.29         10.55   
71   Nintendo      4.36      1.71      3.00         0.23          9.30   
76   Nintendo      3.54      1.24      3.81         0.18          8.76   
137    Capcom      2.47      0.83      2.87         0.12          6.30   

     Critic_Score  Critic_Count User_Score  User_Count Developer Rating  
18            NaN           NaN        NaN         NaN       NaN    NaN  
56            NaN           NaN        NaN         NaN       NaN    NaN  
71            NaN           NaN        NaN         NaN       NaN    NaN  
76            NaN           NaN        NaN         NaN       NaN    NaN  
137           NaN           NaN        NaN         NaN       NaN    NaN

### 7. Find the five publishers with the highest total global sales. Note: You will need to calculate the total global sales for each publisher to do this.

In [49]:
## Create dataframe with only 'Publisher', 'Global_Sales'.
## Calculate the sum of global sales and sort the data by 'Global_Sales' in descending order.
## Display the top 5 rows using the head function.

VGames_GS_df = Video_Games_df[['Publisher', 'Global_Sales']]
VGames_GS_sort_df = VGames_GS_df.groupby(['Publisher']).sum().sort_values(by='Global_Sales', ascending=False)
VGames_GS_sort_df.head(5)


Global_Sales
Publisher                                
Nintendo                          1788.81
Electronic Arts                   1116.96
Activision                         731.16
Sony Computer Entertainment        606.48
Ubisoft                            471.61

### 8. Create a new column in the data frame that calculates the percentage of global sales from North America. Display the first five rows of the new data frame.

In [50]:
## Creating new column called GlobalSales_%tage_NA
## Calculating the percantage of NA sales by dividing the NA_Sales by Global Sales and multiplying by 100.

Video_Games_df["GlobalSales_%tage_NA"] = (Video_Games_df["NA_Sales"] / Video_Games_df["Global_Sales"]*100)
Video_Games_df.head(5)

Name Platform  Year_of_Release         Genre Publisher  \
0                Wii Sports      Wii           2006.0        Sports  Nintendo   
1         Super Mario Bros.      NES           1985.0      Platform  Nintendo   
2            Mario Kart Wii      Wii           2008.0        Racing  Nintendo   
3         Wii Sports Resort      Wii           2009.0        Sports  Nintendo   
4  Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  Critic_Score  \
0     41.36     28.96      3.77         8.45         82.53          76.0   
1     29.08      3.58      6.81         0.77         40.24           NaN   
2     15.68     12.76      3.79         3.29         35.52          82.0   
3     15.61     10.93      3.28         2.95         32.77          80.0   
4     11.27      8.89     10.22         1.00         31.37           NaN   

   Critic_Count User_Score  User_Count Developer Rating  GlobalSales_%tage_NA  
0          51.0          8       322.0  Nintendo      E             50.115110  
1           NaN        NaN         NaN       NaN    NaN             72.266402  
2          73.0        8.3       709.0  Nintendo      E             44.144144  
3          73.0          8       192.0  Nintendo      E             47.635032  
4           NaN        NaN         NaN       NaN    NaN             35.926044

### 9. Find the number NaN entries (missing data values) in each column.

In [51]:
## Using isna() function to count the number of NAN for each columns

Video_Games_df.isna().sum()

Name                       2
Platform                   0
Year_of_Release          269
Genre                      2
Publisher                 54
NA_Sales                   0
EU_Sales                   0
JP_Sales                   0
Other_Sales                0
Global_Sales               0
Critic_Score            8582
Critic_Count            8582
User_Score              6704
User_Count              9129
Developer               6623
Rating                  6769
GlobalSales_%tage_NA       0
dtype: int64

### 10. Try to calculate the median user score of all the video games. You will likely run into an error because some of the user score entries are a non-numerical string that cannot be converted to a float. Find and replace this string with NaN and then calculate the median. Then, replace all NaN entries in the user score column with the median value.

In [52]:
## Using median function on User_Score column, this will give us error due to non numerical string values.

Video_Games_df["User_Score"].median()

TypeError: could not convert string to float: 'tbd'

##### We can see the median function failed on Under_score column due to presence of non numerical values and error also suggest that string value cannot be converted to float. Let convert the non numerical values to NAN.

In [53]:
#### Replacing the non numerical values to NAN.

VG_median_df = Video_Games_df.replace('tbd', np.nan)
VG_median_df.head(10)

Name Platform  Year_of_Release         Genre  \
0                 Wii Sports      Wii           2006.0        Sports   
1          Super Mario Bros.      NES           1985.0      Platform   
2             Mario Kart Wii      Wii           2008.0        Racing   
3          Wii Sports Resort      Wii           2009.0        Sports   
4   Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing   
5                     Tetris       GB           1989.0        Puzzle   
6      New Super Mario Bros.       DS           2006.0      Platform   
7                   Wii Play      Wii           2006.0          Misc   
8  New Super Mario Bros. Wii      Wii           2009.0      Platform   
9                  Duck Hunt      NES           1984.0       Shooter   

  Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  \
0  Nintendo     41.36     28.96      3.77         8.45         82.53   
1  Nintendo     29.08      3.58      6.81         0.77         40.24   
2  Nintendo     15.68     12.76      3.79         3.29         35.52   
3  Nintendo     15.61     10.93      3.28         2.95         32.77   
4  Nintendo     11.27      8.89     10.22         1.00         31.37   
5  Nintendo     23.20      2.26      4.22         0.58         30.26   
6  Nintendo     11.28      9.14      6.50         2.88         29.80   
7  Nintendo     13.96      9.18      2.93         2.84         28.92   
8  Nintendo     14.44      6.94      4.70         2.24         28.32   
9  Nintendo     26.93      0.63      0.28         0.47         28.31   

   Critic_Score  Critic_Count User_Score  User_Count Developer Rating  \
0          76.0          51.0          8       322.0  Nintendo      E   
1           NaN           NaN        NaN         NaN       NaN    NaN   
2          82.0          73.0        8.3       709.0  Nintendo      E   
3          80.0          73.0          8       192.0  Nintendo      E   
4           NaN           NaN        NaN         NaN       NaN    NaN   
5           NaN           NaN        NaN         NaN       NaN    NaN   
6          89.0          65.0        8.5       431.0  Nintendo      E   
7          58.0          41.0        6.6       129.0  Nintendo      E   
8          87.0          80.0        8.4       594.0  Nintendo      E   
9           NaN           NaN        NaN         NaN       NaN    NaN   

   GlobalSales_%tage_NA  
0             50.115110  
1             72.266402  
2             44.144144  
3             47.635032  
4             35.926044  
5             76.668870  
6             37.852349  
7             48.271093  
8             50.988701  
9             95.125397

In [54]:
## Lets calculate the median now to see if it work.

median_vg = VG_median_df["User_Score"].median()
print("The median values for user score is : {}".format(median_vg))

The median values for user score is : 7.5


In [56]:
## Now we have median value 7.5. Lets replace the NAN values with median value 

VG_median_df['User_Score'].fillna(value=VG_median_df['User_Score'].median(), inplace=True)
VG_median_df.head(10)

Name Platform  Year_of_Release         Genre  \
0                 Wii Sports      Wii           2006.0        Sports   
1          Super Mario Bros.      NES           1985.0      Platform   
2             Mario Kart Wii      Wii           2008.0        Racing   
3          Wii Sports Resort      Wii           2009.0        Sports   
4   Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing   
5                     Tetris       GB           1989.0        Puzzle   
6      New Super Mario Bros.       DS           2006.0      Platform   
7                   Wii Play      Wii           2006.0          Misc   
8  New Super Mario Bros. Wii      Wii           2009.0      Platform   
9                  Duck Hunt      NES           1984.0       Shooter   

  Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  \
0  Nintendo     41.36     28.96      3.77         8.45         82.53   
1  Nintendo     29.08      3.58      6.81         0.77         40.24   
2  Nintendo     15.68     12.76      3.79         3.29         35.52   
3  Nintendo     15.61     10.93      3.28         2.95         32.77   
4  Nintendo     11.27      8.89     10.22         1.00         31.37   
5  Nintendo     23.20      2.26      4.22         0.58         30.26   
6  Nintendo     11.28      9.14      6.50         2.88         29.80   
7  Nintendo     13.96      9.18      2.93         2.84         28.92   
8  Nintendo     14.44      6.94      4.70         2.24         28.32   
9  Nintendo     26.93      0.63      0.28         0.47         28.31   

   Critic_Score  Critic_Count User_Score  User_Count Developer Rating  \
0          76.0          51.0          8       322.0  Nintendo      E   
1           NaN           NaN        7.5         NaN       NaN    NaN   
2          82.0          73.0        8.3       709.0  Nintendo      E   
3          80.0          73.0          8       192.0  Nintendo      E   
4           NaN           NaN        7.5         NaN       NaN    NaN   
5           NaN           NaN        7.5         NaN       NaN    NaN   
6          89.0          65.0        8.5       431.0  Nintendo      E   
7          58.0          41.0        6.6       129.0  Nintendo      E   
8          87.0          80.0        8.4       594.0  Nintendo      E   
9           NaN           NaN        7.5         NaN       NaN    NaN   

   GlobalSales_%tage_NA  
0             50.115110  
1             72.266402  
2             44.144144  
3             47.635032  
4             35.926044  
5             76.668870  
6             37.852349  
7             48.271093  
8             50.988701  
9             95.125397

In [57]:
## Lets check if under_score column has any NAN values.

count_NAN = VG_median_df["User_Score"].isna().sum()
print("The number of NAN values in Under_Score column in dataframe is: {}".format(count_NAN))

The number of NAN values in Under_Score column in dataframe is: 0
